src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -474,7 +474,7 @@ protected DynamoModel(IStartConfiguration config)
             MigrationManager.MigrationTargets.Add(typeof(WorkspaceMigrations));
 
             var thread = config.SchedulerThread ?? new DynamoSchedulerThread();
-            Scheduler = new DynamoScheduler(thread, IsTestMode ? TaskProcessMode.Synchronous : TaskProcessMode.Asynchronous);
+            Scheduler = new DynamoScheduler(thread, config.ProcessMode);
```

Comment:
Wasn't aware that `ProcessMode` is already in `IStartConfiguration`, curious to know why wasn't it used before, maybe @ke-yu can fill me in. Apart from that, if we are sure all callers check for `IsTestMode` to determine the value of `ProcessMode`, then this looks good!